### PR TITLE
Add schedule to start task metric

### PR DESCRIPTION
**Pull request by Ardagan**
**What changed?**
Added task schedule to start latency

**Why?**
Improve observability.

**How did you test it?**
automated/manual tests.

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

**Comment by paulnpdev** [comment]
reminder to make sure other (non-code) documentation gets updated for this
too, as well as communicated to the product team.

File: common/metrics/defs.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ const (
 	StatsTypeTagName   = "stats_type"
 	CacheTypeTagName   = "cache_type"
 	FailureTagName     = "failure"
+	TypeTagName        = "type"
```
Comment (yiminc) [inline common/metrics/defs.go:87]: type is too generic. Use TaskTypeTagName = "task_type"

File: common/metrics/tags.go [hunk 2]
```diff
@@ -96,6 +98,10 @@ type (
 	failureTag struct {
 		value string
 	}
+
+	typeTag struct {
```
Comment (yiminc) [inline common/metrics/tags.go:102]: use taskTypeTag

**Review by yiminc — APPROVED** [review]

File: common/metrics/defs.go (hunk 1 above)
Comment (Ardagan) [reply common/metrics/defs.go:87]: I think that it's better to have generic tags, so they can be reused in other metrics.

**Comment by Ardagan** [comment]
https://github.com/temporalio/temporal/issues/1754

**Pull request merged**